### PR TITLE
adafruit_ble: support chibios

### PR DESCRIPTION
**Pull request by Gigahawk**
Enable ADAFRUIT_BLE  on ChibiOS.

## Description

Tested on a Bluefruit LE SPI with an STM32F405, it seems to mostly work.
It seems to work fine I pair my phone over Bluetooth, if I power cycle the board, then I can reconnect as normal but no keystrokes get sent even after `OUT_BT` is pressed.
If I unpair and then pair again it works until the next power cycle.
Disconnecting from my phone without turning off the board works fine.

EDIT: Apparently the latest Bluefruit firmware is just buggy, a cursory test of 0.6.7 as suggested from https://github.com/adafruit/Adafruit_BluefruitLE_Firmware/issues/27 seems to work fine, although it causes one of the QMK init commands to fail with `  > result: ERROR`.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [x] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: tmk_core/protocol/chibios/usb_main.c [hunk 1]
```diff
@@ -109,7 +118,7 @@ static const USBDescriptor *usb_get_descriptor_cb(USBDriver *usbp, uint8_t dtype
     static USBDescriptor desc;
     uint16_t             wValue = ((uint16_t)dtype << 8) | dindex;
     desc.ud_string              = NULL;
-    desc.ud_size                = get_usb_descriptor(wValue, wIndex, (const void **const) & desc.ud_string);
+    desc.ud_size                = get_usb_descriptor(wValue, wIndex, (const void **const)&desc.ud_string);
```
Comment (Gigahawk) [inline tmk_core/protocol/chibios/usb_main.c:121]: This is just what QMK MSYS does when I commit.
Has there been a change between the CI and the git hook?

**Comment by fauxpark** [comment]
I'm already working on this.

**Comment by nklue** [comment]
@Gigahawk Just found an issue, while working on setting this up. This [16127](https://github.com/qmk/qmk_firmware/pull/16127) looks to be merged and renames all of the Adafruit functions, vars, etc. Not sure if there's additional consideration required but thought I'd mention as I'm working on testing this now.

**Comment by Gigahawk** [comment]
everything has been renamed, should work fine now

**Comment by Gigahawk** [comment]
@fauxpark is there anything actually preventing this from being merged?

I know the bluetooth refactor is coming eventually, but it seems like that's going to take a while.
I don't see why this can't be merged in the meantime given how small of a change it is.

**Comment by rockybulwinkle** [comment]
@Gigahawk I tried your changes but get an undefined file, "outputselect.h". Can you provide any insight? I'm using the teensy 4.1.

**Comment by rockybulwinkle** [comment]
Also, @fauxpark, is your work being tracked by an issue and/or on a feature branch? I cannot find it.

**Pull request closed without merging**